test/Microsoft.AspNetCore.Server.KestrelTests/PipelineExtensionTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests
+{
+    public class PipelineExtensionTests : IDisposable
+    {
+        // ulong.MaxValue.ToString().Length
+        private const int _ulongMaxValueLength = 20; 
+
+        private readonly IPipe _pipe;
+        private readonly PipeFactory _pipeFactory = new PipeFactory();
+
+        public PipelineExtensionTests()
+        {
+            _pipe = _pipeFactory.Create();
+        }
+
+        [Theory]
+        [InlineData(ulong.MinValue)]
+        [InlineData(ulong.MaxValue)]
+        [InlineData(4_8_15_16_23_42)]
```

Comment:
I wanted to test a random number somewhere between 0 and ulong.MaxValue. http://lost.wikia.com/wiki/The_Numbers